### PR TITLE
Feature/versioning

### PR DESCRIPTION
**Pull request by ValeSayfa**
**Proposed Changes**

- Adds revision selection to general views of algorithms and implementations

### CODE REVIEW

**Review by manuwei — APPROVED** [review]
LGTM, I only added some minor changes regarding the reformatting of the generated files + updated the version

File: src/app/components/algorithms/algorithm-view/algorithm-view.component.ts [hunk 1]
```diff
@@ -293,12 +309,66 @@ export class AlgorithmViewComponent implements OnInit, OnDestroy {
...
+  getRevision(revision: EntityModelRevisionDto): void {
+    this.algorithmService
+      .getAlgorithmRevision({
+        algorithmId: this.algorithm.id,
+        revisionId: revision.id,
+      })
+      .subscribe(
+        (algorithmRevision) => {
+          this.frontendAlgorithm = algorithmRevision;
+          let subheading = this.frontendAlgorithm.computationModel
```
Comment (lharzenetter) [inline src/app/components/algorithms/algorithm-view/algorithm-view.component.ts:327]: this code here is exaclty the same as in https://github.com/UST-QuAntiL/qc-atlas-ui/pull/123/files#diff-5bfc3b3483dc91598e30219e090f18123a4d464dbb63836c21d1d7eac1948cb0R290
I'd recommend refactoring it into a utility method.

**Review by lharzenetter — CHANGES_REQUESTED** [review]
besides the code duplication, LGTM

File: src/app/components/algorithms/algorithm-view/algorithm-view.component.ts (hunk 1 above)
Comment (ValeSayfa) [reply src/app/components/algorithms/algorithm-view/algorithm-view.component.ts]: The mentioned code has been refactored as suggested

**Review by lharzenetter — APPROVED** [review]

**Review by lharzenetter — CHANGES_REQUESTED** [review]
Please run the generation of the UI model again such that they are not changed completely. I guess @manuwei can support here.

File: package-lock.json [hunk 2]
```diff
@@ -1,19342 +1,8 @@
 {
   "name": "qc-atlas-ui",
   "version": "0.0.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
Comment (lharzenetter) [inline package-lock.json:4]: Why did you remove the packages?

File: src/app/components/algorithms/algorithm-properties/algorithm-properties.component.html [hunk 3]
```diff
@@ -143,48 +151,51 @@
...
+           <app-select-input name="Sketch type"
+                             [value]="algorithm.sketches"
+                             [includeEmpty]="true"
+                             [choices]="availableSketchOptions"
+                             (onSaveChanges)="onChangesSaved($event, 'sketch')"></app-select-input>
+         </mat-card>
+       </div>-->
     </div>
-
-    <!-- <div class="col-lg-6 col-sm-12">
```
Comment (lharzenetter) [inline src/app/components/algorithms/algorithm-properties/algorithm-properties.component.html:180]: What's with this code? I guess this can be removed here? Or, @manuwei @salmma, was this intentional?

**Review by lharzenetter — COMMENTED** [review]
Besides minor questions, LGTM

**Comment by manuwei** [comment]
@lharzenetter mentioned using ``ng postgen`` to reformat the generated code

**Review by lharzenetter — APPROVED** [review]
